packages/shorebird_cli/lib/src/commands/run_command.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:archive/archive_io.dart';
+import 'package:args/command_runner.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/auth/auth.dart';
+import 'package:shorebird_cli/src/command_runner.dart';
+import 'package:shorebird_cli/src/compute.dart';
+import 'package:shorebird_code_push_api_client/shorebird_code_push_api_client.dart';
+
+typedef StartProcess = Future<Process> Function(
+  String executable,
+  List<String> arguments, {
+  bool runInShell,
+});
+
+/// {@template run_command}
+///
+/// `shorebird run`
+/// Run the Flutter application.
+/// {@endtemplate}
+class RunCommand extends Command<int> {
+  /// {@macro run_command}
+  RunCommand({
+    required Auth auth,
+    required ShorebirdCodePushApiClientBuilder codePushApiClientBuilder,
+    required Logger logger,
+    StartProcess? startProcess,
+  })  : _auth = auth,
+        _buildCodePushApiClient = codePushApiClientBuilder,
+        _logger = logger,
+        _startProcess = startProcess ?? Process.start;
+
+  @override
+  String get description => 'Run the Flutter application.';
+
+  @override
+  String get name => 'run';
+
+  final Auth _auth;
+  final ShorebirdCodePushApiClientBuilder _buildCodePushApiClient;
+  final Logger _logger;
+  final StartProcess _startProcess;
+
+  @override
+  Future<int> run() async {
+    final session = _auth.currentSession;
+    if (session == null) {
+      _logger
+        ..err('You must be logged in to run.')
+        ..err("Run 'shorebird login' to log in and try again.");
+      return ExitCode.noUser.code;
+    }
+
+    final shorebirdEnginePath = p.join(
+      Directory.current.path,
+      '.shorebird',
```

Comment:
I might leave a comment that this ends up as part of some shorebird install eventually (each flutter checkout probably doesn't cache its own engine).

Reply:
yup good call 👍